feat: Make gemini accept the openai parameter parallel_tool_calls

aholmberg opened the pull request:
When mapping, allow the parameter: True because that is the intrinsic behavior of Gemini. Allow False, but reject if there are multiple tools because there's no actual equivalent in Gemini.

fixes #9686

ref: issues/9686

## Title

<!-- e.g. "Implement user authentication feature" -->

## Relevant issues
fixes #9686

## Pre-Submission checklist

**Please complete all items before asking a LiteLLM maintainer to review your PR**

- [x] I have Added testing in the [`tests/litellm/`](https://github.com/BerriAI/litellm/tree/main/tests/litellm) directory, **Adding at least 1 test is a hard requirement** - [see details](https://docs.litellm.ai/docs/extras/contributing_code)
- [x] I have added a screenshot of my new test passing locally 
![image](https://github.com/user-attachments/assets/c6aa49eb-4a62-41e9-8035-a0ebd4da84c9)

- [x] My PR passes all unit tests on [`make test-unit`](https://docs.litellm.ai/docs/extras/contributing_code)
- [x] My PR's scope is as isolated as possible, it only solves 1 specific problem


## Type
🆕 New Feature
🐛 Bug Fix

## Changes
When mapping, allow the parameter: True because that is the intrinsic behavior of Gemini. Allow False, but reject if there are multiple tools because there's no actual equivalent in Gemini.

aholmberg committed:
feat: Make gemini accept the openai parameter parallel_tool_calls

When mapping, allow the parameter: True because that is the
intrinsic behavior of Gemini. Allow False, but reject if there
are multiple tools because there's no actual equivalent in Gemini.

fixes #9686

ref: issues/9686

krrishdholakia commented:
Hey @aholmberg if you rebase with main, it should be fixed

krrishdholakia commented on tests/llm_translation/test_vertex.py:
move this test inside `tests/litellm` so it can run on the github action (llm_translation tests are mixed with real api calls, so don't run on contributor pr's)

krrishdholakia commented on litellm/llms/vertex_ai/gemini/vertex_and_google_ai_studio_gemini.py:
```diff
@@ -463,6 +464,19 @@ def map_openai_params(
+                if value is False:
+                    tools = non_default_params.get("tools", non_default_params.get("functions"))
+                    num_function_declarations = len(tools) if isinstance(tools, list) else 0
+                    if num_function_declarations > 1:
```
add support for user setting either `completion(...,drop_params = True)` or `litellm.drop_params = True`

if so - i assume we would just not set this flag `parallel_tool_calls=False`

aholmberg replied:
I'm not sure what is meant by this comment. The way I was thinking about it, parallel tool calls `True` is the default. If it's set `True` we don't need to inject anything. This new logic is just bailing if it's being set `False` and more than one tool is present (no way I know of to emulate that in Gemini).

aholmberg committed:
chore: cleanup and move test_vertex.py down to tests/litellm

as suggested in https://github.com/BerriAI/litellm/pull/11125#discussion_r2105905871

aholmberg commented:
Rebased and moved the test.
